### PR TITLE
Add support for oauth authorization

### PR DESCRIPTION
**Pull request by strausr**
### Brief Summary of Changes
Add support for oauth authorization API

#### What Does This PR Address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [x] New feature
- [ ] Bug fix
- [ ] Adds more tests

#### Are Tests Included?
- [x] Yes
- [ ] No

### CODE REVIEW

File: lib/utils/ensureOption.js [hunk 1]
```diff
@@ -8,8 +8,7 @@
 function defaults(defaultOptions) {
   return function ensureOption(options, name, defaultValue) {
     let value;
-
-    if (typeof options[name] !== 'undefined') {
+    if (typeof options[name] !== 'undefined' || name === 'oauth_token') {
```
Comment (patrick-tolosa) [inline lib/utils/ensureOption.js:11]: What does this do specifically?

File: test/integration/api/authorization/oAuth_authorization_spec.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      api_secret: "1234",
+      oauth_token: null
+    });
+    return helper.provideMockObjects((mockXHR, writeSpy, requestSpy) => {
+      cloudinary.v2.uploader.upload(PUBLIC_ID)
+      return sinon.assert.calledWith(requestSpy, sinon.match({ auth: null }));
+    });
+  });
+
+  it("should fail when missing all credentials (upload_api)", function() {
```
Comment (patrick-tolosa) [inline test/integration/api/authorization/oAuth_authorization_spec.js:107]: Maybe adjust the wording, we're not testing for failure here exactly right?

We're testing that auth is null (same for a test or two above)
what do you think?

**Review by patrick-tolosa — APPROVED** [review]

File: lib/utils/ensureOption.js (hunk 1 above)
Comment (strausr) [reply lib/utils/ensureOption.js:11]: this sets 'oauth_token' if it not undefined

**Review by const-cloudinary — APPROVED** [review]
Looks good!
Please add one more test for unsigned upload

File: test/integration/api/authorization/oAuth_authorization_spec.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    expect(() => {
+      cloudinary.v2.uploader.upload(PUBLIC_ID)
+    }).to.throwError(/Must supply api_key/);
+  });
+
+  it("should not need credentials for unsigned upload", function() {
+    cloudinary.config({
+      api_key: undefined,
+      api_secret: undefined,
+      oauth_token: '1234'
```
Comment (const-cloudinary) [inline test/integration/api/authorization/oAuth_authorization_spec.js:121]: `oauth_token` should be also `undefined`

**Review by const-cloudinary — CHANGES_REQUESTED** [review]

File: test/integration/api/authorization/oAuth_authorization_spec.js (hunk 3 above)
Comment (strausr) [reply test/integration/api/authorization/oAuth_authorization_spec.js]: done

**Review by const-cloudinary — APPROVED** [review]

**Pull request merged**